sdk/src/main/scala/com/horizen/SidechainNodeViewHolder.scala

```diff
@@ -157,23 +158,22 @@ class SidechainNodeViewHolder(sidechainSettings: SidechainSettings,
       (history, wallet)
   }
 
-  // Check is the modifier ends the withdrawal epoch, so notify History and Wallet about fees to be payed.
-  // Scan modifier by the Wallet considering the forger fee payments.
-  override protected def scanBlockWithFeePayments(history: HIS, state: MS, wallet: VL, modToApply: SidechainBlock): (HIS, VL) = {
-    val stateWithdrawalEpochNumber: Int = state.getWithdrawalEpochInfo.epoch
-    if (state.isWithdrawalEpochLastIndex) {
-      val feePayments = state.getFeePayments(stateWithdrawalEpochNumber)
-      val historyAfterUpdateFee = history.updateFeePaymentsInfo(modToApply.id, SidechainFeePaymentsInfo(feePayments))
-
-      val walletAfterApply: VL = wallet.scanPersistent(modToApply, stateWithdrawalEpochNumber, feePayments, Some(state))
+  override def getFeePaymentsInfo(state: MS, epochNumber: Int) : FPI = {
+    val feePayments = state.getFeePayments(epochNumber)
+    SidechainFeePaymentsInfo(feePayments)
+  }
 
-      (historyAfterUpdateFee, walletAfterApply)
-    } else {
-      val walletAfterApply: VL = wallet.scanPersistent(modToApply, stateWithdrawalEpochNumber, Seq(), None)
-      (history, walletAfterApply)
+  def getScanPersistentWallet(modToApply: SidechainBlockBase[_ <: Transaction, _ <: SidechainBlockHeaderBase], stateOp: Option[MS], epochNumber: Int, wallet: VL) : VL = {
```

Comment:
1) add `override`
2)`SidechainBlockBase[_ <: Transaction, _ <: SidechainBlockHeaderBase]` -> `SidechainBlock` . So, no need to case `.asInstanceOf[SidechainBlock]`.
same for account model

Reply:
Done